ow_power_system/src/power_system_node.cpp

```diff
@@ -172,7 +190,7 @@ void PowerSystemNode::jointStatesCb(const sensor_msgs::JointStateConstPtr& msg)
   m_mechanical_power_raw_pub.publish(mechanical_power_raw_msg);
   m_mechanical_power_avg_pub.publish(mechanical_power_avg_msg);
 
-  m_unprocessed_mechanical_power += power_watts;
+  m_unprocessed_mechanical_power = mean_mechanical_power;
```

Comment:
Based on the `/joint_states` publish rate of 50Hz, the GSAP prognoser rate of 0.5 Hz, and a moving average window size of 10 datapoints, the prognoser only ever sees a mean mechanical power that's calculated from the 10 most recent datapoints out of the 25 new datapoints available to it each cycle. I am wondering if it would be more preferable to compute the mean for all 25 new datapoints?

The easiest way to achieve this is to change the moving average window size to 25.

Reply:
Great catch!  Your reasoning makes sense to me and I've changed the window to 25.